File: etsyCatalogs/etsyCatalogs/View/DetailViewController.m
//
//  DetailViewController.m
//  etsyCatalogs
//
//  Created by Ihonahan Buitrago on 9/1/14.
//  Copyright (c) 2014 Ihonahan Buitrago. All rights reserved.
//

#import "DetailViewController.h"

@interface DetailViewController ()

@end

@implementation DetailViewController
            
#pragma mark - Managing the detail item


- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}


- (void)setCatalogDetail:(CatalogItem *)theCatalog
{
    self.detailItem = theCatalog;
    
    self.titleLabel.text = self.detailItem.itemTitle;
    self.priceTitleLabel.text = NSLocalizedString(@"Price: $", nil);
    self.priceLabel.text = [NSString stringWithFormat:@"%f", self.detailItem.price];
    self.descriptionView.text = self.detailItem.itemDescription;
    if (self.detailItem.images.count) {
        CatalogListingImage *image0 = [self.detailItem.images objectAtIndex:0];
        double scale = 1.0;
        if ([[UIScreen mainScreen] respondsToSelector:@selector(scale)] == YES) {
            scale = [[UIScreen mainScreen] scale];
        }
        double w = image0.fullWidth / scale;
        double h = image0.fullHeight / scale;
        
        double imgW = self.imageView.frame.size.width;
        double imgH = (h * imgW) / w;
        
        self.imageView.frame = CGRectMake(self.imageView.frame.origin.x,
                                          self.imageView.frame.origin.y,
                                          imgW ,
                                          imgH);
        [self.imageView setImageWithURL:[NSURL URLWithString:image0.urlFullSize]];
    }
}



@end

File: etsyCatalogsSwift/etsyCatalogsSwift/Controller/DetailViewController.swift
//
//  DetailViewController.swift
//  etsyCatalogsSwift
//
//  Created by Ihonahan Buitrago on 9/19/14.
//  Copyright (c) 2014 Ihonahan Buitrago. All rights reserved.
//

import UIKit

class DetailViewController: UIViewController {

    @IBOutlet weak var titleLabel: UILabel!
    @IBOutlet weak var priceTitleLabel: UILabel!
    @IBOutlet weak var priceLabel: UILabel!
    @IBOutlet weak var descriptionView: UILabel!

    var detailItem : CatalogItem = CatalogItem()


    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

    
    func configureDetailView()
    {
        self.titleLabel.text = self.detailItem.itemTitle
        self.priceTitleLabel.text = NSLocalizedString("Price: $", comment: "")
        self.priceLabel.text = String(format: "%.2f", self.detailItem.price!)
        self.descriptionView.text = self.detailItem.itemDescription
        
        
    }

}

